pkg/nvidia/Dockerfile

```diff
@@ -0,0 +1,102 @@
+
+# Jetson Linux tarball
+ENV JETSON_LINUX=https://developer.nvidia.com/downloads/embedded/l4t/r35_release_v5.0/release/jetson_linux_r35.5.0_aarch64.tbz2
+
+ADD ${JETSON_LINUX} /jetson_linux.tbz2
+
+# We need to build dpkg in order to get zstd compression support
+ENV DPKG_REVISION=1.22.6
+ADD --keep-git-dir=true https://salsa.debian.org/dpkg-team/dpkg.git#${DPKG_REVISION} /dpkg
+WORKDIR /dpkg
+RUN ./autogen && \
+    ./configure --prefix=/ && \
+    make -j "$(nproc)" && \
+    make install
+
+# CDI Files
+COPY cdi /cdi
+
+# Extract tarball
+WORKDIR /
+RUN tar -xjf /jetson_linux.tbz2
+
+# Build nvidia-ctk
+ENV NVIDIA_CONTAINER_TOOLKIT_REV=v1.16.0
+ADD --keep-git-dir=true https://github.com/NVIDIA/nvidia-container-toolkit.git#${NVIDIA_CONTAINER_TOOLKIT_REV} /nvct
+
+WORKDIR /nvct
+
+COPY patches/* /nvct
+RUN for x in /nvct/*.patch; do \
+        git am "$x" ; \
+    done
+
+RUN mkdir -p dist
+RUN make cmd-nvidia-ctk && cp nvidia-ctk dist/
+RUN make cmd-nvidia-cdi-hook && cp nvidia-cdi-hook dist/
+
+# Get and extract ldconfig for glibc
+ADD http://ports.ubuntu.com/ubuntu-ports/pool/main/g/glibc/libc-bin_2.31-0ubuntu9.16_arm64.deb /
+RUN mkdir -p /ldconfig-bin && \
+    dpkg -x /libc-bin_2.31-0ubuntu9.16_arm64.deb /ldconfig && \
+    cp /ldconfig/sbin/ldconfig.real /ldconfig-bin/ldconfig-glibc
+
+
+FROM build-${TARGETARCH} as build
+
+# Copy udev rules
+COPY udev/rules.d/* /rules.d/
+
+# Copy scripts to processing CDI files and initialization
+COPY scripts/* /
+
+# Process CDI files
+WORKDIR /
+RUN mkdir -p /ldconfig-bin /nvfanctrl/dist /nvct/dist /cdi /rootfs-dist /cdi-dist /rules.d && \
+    mkdir -p /rootfs && \
+    if [ -n "$(ls /cdi/* 2> /dev/null)" ]; then \
+        for x in /cdi/*.yaml; do \
+            OUTFILE=/cdi-dist/$(basename "$x") ; \
+            /process-cdi.sh "$x" /rootfs /rootfs-dist &&  \
+            yq '. | (.devices[].containerEdits.hooks[] | select(.path == "/usr/bin/nvidia-ctk")).path = "/opt/vendor/nvidia/bin/nvidia-ctk"' -i "$x" && \
+            yq '. | (.devices[].containerEdits.hooks[] | select(.path == "/usr/bin/nvidia-cdi-hook")).path = "/opt/vendor/nvidia/bin/nvidia-cdi-hook"' -i "$x" && \
+            /nvct/dist/nvidia-ctk cdi transform root --relative-to host \
+                --from /usr --to /opt/vendor/nvidia/dist/usr --input "$x" | \
+            /nvct/dist/nvidia-ctk cdi transform root --relative-to host \
+                --from /etc --to /opt/vendor/nvidia/dist/etc | \
+            /nvct/dist/nvidia-ctk cdi transform root --relative-to host \
+                --from /lib --to /opt/vendor/nvidia/dist/lib --output "$OUTFILE" && \
+            yq '. | (.devices[].containerEdits.hooks[] | select(.args[] == "update-ldcache")).args += ["--ldconfig-path", "/opt/vendor/nvidia/bin/ldconfig-glibc"]' -i "$OUTFILE" ; \
+        done ; \
+    fi && \
+    rm -rf /rootfs/*
+
+FROM scratch
+COPY --from=build /cdi-dist/*.yaml /etc/cdi/
+COPY --from=build /rules.d/* /opt/vendor/nvidia/etc/udev/rules.d/
```

Comment:
In most other packages, we have the build image install lots of things to `/out/` and then have a single line in 

```dockerfile
FROM scratch
COPY --from=build /out /
```

Would that work here?

Reply:
Probably, but I'm just following the same approach of many other packages, like kvm-tools, bsp-imx, newlog, bpftrace, gpt-tools, udev, wwan, etc... so I would let as it is...